mitxpro/utils.py

```diff
@@ -603,3 +603,18 @@ def get_js_settings(request: HttpRequest):
         "enable_taxes_display": settings.FEATURES.get("ENABLE_TAXES_DISPLAY", False),
         "enable_enterprise": settings.FEATURES.get("ENABLE_ENTERPRISE", False),
     }
+
+
+def clean_url(url, *, trim_spaces=True, remove_query_params=False):
+    """
+    Cleans a URL by removing the extra spaces and Optionally removes the query params to return the base URL.
+    """
+    if not url:
+        return ""
+
+    if trim_spaces:
+        url = url.strip()
+
+    if remove_query_params:
+        url = url[: url.find("?")]
+    return url.strip()
```

Comment:
I think this can be replaced by urllib.parse functions?

Reply:
There are a few bad URLs in the data that urllib.parse functions cannot handle. 
<img width="1715" alt="Screenshot 2024-06-21 at 8 39 20 PM" src="https://github.com/mitodl/mitxpro/assets/52656433/a9b59122-71f5-451f-9a23-c9638e0fa103">